src/uri.rs

```diff
@@ -126,8 +124,12 @@ impl Quoter {
                                 idx += 1;
                                 continue;
                             }
+
+                            buf.extend_from_slice(&pct);
```

Comment:
Is this code supposed to be reached?

Reply:
yes it can be reached, it's the default case when we don't want to decode the character. I removed all the dead code, it should be clearer